src/main.rs

```diff
@@ -0,0 +1,100 @@
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+use clap::{arg, Parser};
+use error::Error;
+use http::HeaderMap;
+use http::{header::AUTHORIZATION, HeaderValue};
+use jsonrpsee::http_client::HttpClientBuilder;
+use jsonrpsee::server::Server;
+use jsonrpsee::RpcModule;
+use server::{EthApiServer, EthEngineApi};
+use tracing::{info, Level};
+use tracing_subscriber::EnvFilter;
+
+mod error;
+mod server;
+
+#[derive(Parser, Debug)]
+#[clap(author, version, about)]
+struct Args {
+    /// JWT token to authenticate communication between the consensus and execution clients
+    #[arg(long, env)]
+    jwt_token: String,
+
+    /// URL of the local l2 execution engine
+    #[arg(long, env)]
+    l2_url: String,
+
+    /// URL of the builder execution engine
+    #[arg(long, env)]
+    builder_url: String,
+
+    /// Host to run the server on
+    #[arg(long, env, default_value = "0.0.0.0")]
+    rpc_host: String,
+
+    /// Port to run the server on
+    #[arg(long, env, default_value = "8081")]
+    rpc_port: u16,
+
+    // Enable tracing
+    #[arg(long, env, default_value = "false")]
+    tracing: bool,
+
+    /// Log level
+    #[arg(long, env, default_value = "info")]
+    log_level: Level,
+}
+
+type Result<T> = core::result::Result<T, Error>;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let args: Args = Args::parse();
+
+    // Initialize logging
+    tracing_subscriber::fmt()
+        .with_env_filter(EnvFilter::new(args.log_level.to_string())) // Set the log level
+        .init();
+
+    // Initialize the local execution engine client
+    let mut headers = HeaderMap::new();
```

Comment:
Did you try this part? I think for making http client requests you need to do [sign the token](https://github.com/ethereum-optimism/op-geth/blob/5f7ebba8a124ae87225f81a1a9c827f8a534f2b7/node/jwt_auth.go#L33) not only include it on the header.